app/controllers/application_controller.rb

```diff
@@ -7,11 +7,44 @@ def render_404
     # DPR: supposedly this will actually render a 404 page in production
     raise ActionController::RoutingError.new('Not Found')
   end
+  protect_from_forgery with: :exception
+
+  before_action :require_login
+
+skip_before_action :require_login, only: [:require_login, :save_and_flash, :find_user]
+
+protected
```

Comment:
Controller filters only apply to controller actions, not to any method defined in a controller. This means you shouldn't need the `skip_before_action` line here, since none of `:require_login`, `:save_and_flash` or `:find_user` are actions.